core/conn/jdbc_type2/native/SqlInterface.cpp

```diff
@@ -246,10 +246,14 @@ SQLRETURN GetJDBCValues( SQLItemDesc_def *SQLItemDesc,
 		SQLItemDesc->signType      = TRUE;
 		break;
 	case SQLTYPECODE_DATETIME:
+		// Calculate the length based on YYYY/MM/DD HH:MM:SS.ffffff
+        SQLItemDesc->maxLen = 20 + 6;
```

Comment:
No need to use the internal format because SQLCLI accepts TIMESTAMP, DATE and TIME in character format. I think the point I am trying to make here is SQLItemDesc->maxLen isused by getMemoryAllocInfo to allocate memory buffer accordingly on the client side to be shipped to the JNI side. If so, I am curious how the parameters for date/time/timestamp worked till now?